### ISSUE TITLE
Tests + preview causes "Something went wrong" - "Status should be present after tests are run"

### ISSUE THREAD

**Issue by antoineol** [codesandbox/sandpack#927]
# Bug report

Hey team, thanks for this amazing project! I've faced an issue while experimenting with it. Thanks!

<img width="404" alt="image" src="https://user-images.githubusercontent.com/6078211/236524221-7f822036-e94c-4b15-8213-f6ddc88641ea.png">

Repro on a codesandbox here: https://codesandbox.io/p/github/antoineol/repro-sandpack-test-issue/

## Packages affected

-   [ ] sandpack-client
-   [x] sandpack-react

## Description of the problem

If tests and preview run simultaneously, after the tests pass, an error is shown in the preview instead of the normal rendering.

**Note:** if the issue does not appear the first time you open my codesandbox, you may need to refresh the project. Then the error should consistently appear on refresh.

<img width="88" alt="image" src="https://user-images.githubusercontent.com/6078211/236525949-b7d5c661-0627-4da1-a849-e0abf1f5a533.png">

## What were you doing when the problem occurred?

Trying to get tests running in addition to the preview.

### What steps can we take to reproduce the problem?

Wait until the test passes. It then updates the preview to a red error page as shown in the screenshot above.

The github repo reproducing the issue: https://github.com/antoineol/repro-sandpack-test-issue
(yarn install && yarn dev)

The changes to reproduce it are in the latest commit, after creating a vite react-ts project.

### Your Environment

| Software                | Name/Version |
| ----------------------- | ------------ |
| Sandpack-client version | 2.6.3 |
| Sandpack-react version  | 2.6.3 |
| Browser                 | Chrome latest version |
| Operating System        | Windows 11 |

**Comment by SSHari** [comment]
This one was a bit tricky to figure out, but it looks like this is due to an upstream issue in the `codesandox-client` repo that we're running into because we dispatch a `compile` event multiple times when the `runtime` client is initialized. I was able to consistently reproduce this issue in Storybook as well. Actually, it doesn't require a `SandpackPreview` for this error to be thrown (the Preview just makes it more obvious).

If you open a Storybook story (I used the Verbose Mode Tests story) then you'll see this error thrown in your dev tools console:

<img width="727" alt="image" src="https://github.com/codesandbox/sandpack/assets/11434879/ea924734-7dd4-4c44-9700-a5e1290520a0">

I've gone ahead and opened a [PR in the codesandbox-client repo](https://github.com/codesandbox/codesandbox-client/pull/7620). This issue can probably be re-verifed once/if the PR makes it into the other repo.
